sw/airborne/peripherals/ak8963.c

```diff
@@ -37,10 +37,11 @@ void ak8963_init(struct Ak8963 *ak, struct i2c_periph *i2c_p, uint8_t addr)
   ak->i2c_p = i2c_p;
   /* set i2c address */
   ak->i2c_trans.slave_addr = addr;
-  ak->i2c_trans.status = I2CTransDone;
+  ak->i2c_trans.status = I2CTransFailed;
```

Comment:
why setting this to Failed then Done at the end of the init ?

Reply:
Its been a wilde ago, but there was someting, dont recall, so will have to do a new longflight > 45 min extended batt to validate.

Reply:
this is the init function, only called once after boot, so I don't see how it is related to long flight ?

Reply:
But there was something like  ak->init_status = AK_CONF_UNINIT; the evet got already called and there it wa not in AK_STATUS_IDLE or AK_STATUS_READ thus to default but there only   I2CTransSuccess or I2CTransFailed make it go to idle...so if not set to I2CTransFailed in the begin it is never reached. I will change and retest, but this was one of the once in so many times cases I remember.

Reply:
how is it possible that `event` functions are been called before and of `init` ? (https://github.com/paparazzi/paparazzi/blob/master/sw/airborne/firmwares/fixedwing/main.c#L45)